### PR TITLE
[Parser] Use the new parser in wasm-shell and wasm-as

### PR DESCRIPTION
**Pull request by tlively**
Updating just one or the other of these tools would cause the tests
spec/import-after-*.fail.wast to fail, since only the updated tool would
correctly fail to parse its contents. To avoid this, update both tools at
once. (The tests erroneously pass before this change because check.py does not
ensure that .fail.wast tests fail, only that failing tests end in .fail.wast.)

In wasm-shell, to minimize the diff, only use the new parser to parse modules
and instructions. Continue using the legacy parsing based on s-expressions for
the other wast commands. Updating the parsing of the other commands to use
`Lexer` instead of `SExpressionParser` is left as future work. The boundary
between the two parsing styles is somewhat hacky, but it is worth it to enable
incremental development.

Update the tests to fix incorrect wast rejected by the new parser. Many of the
spec/old_* tests use non-standard forms from before Wasm MVP was standardized,
so fixing them would have been onerous. All of these tests have non-old_*
variants, so simply delete them.

### CODE REVIEW

**Comment by tlively** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6529?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6529** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6529?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6528** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6528?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tlively and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/spec/simd.wast [hunk 1]
```diff
@@ -1050,8 +1050,8 @@
 )
 
 ;; conversions
-(assert_return (invoke "i32x4.trunc_sat_f32x4_s" (v128.const f32x4 42 nan infinity -infinity)) (v128.const i32x4 42 0 2147483647 -2147483648))
-(assert_return (invoke "i32x4.trunc_sat_f32x4_u" (v128.const f32x4 42 nan infinity -infinity)) (v128.const i32x4 42 0 4294967295 0))
+(assert_return (invoke "i32x4.trunc_sat_f32x4_s" (v128.const f32x4 42 nan inf -inf)) (v128.const i32x4 42 0 2147483647 -2147483648))
+(assert_return (invoke "i32x4.trunc_sat_f32x4_u" (v128.const f32x4 42 nan inf -inf)) (v128.const i32x4 42 0 4294967295 0))
```
Comment (kripken) [inline test/spec/simd.wast:1054]: Lot of SIMD changes... did we write this by hand, or did the upstream spec change in the middle, I'm just curious?

**Review by kripken — APPROVED** [review]

File: test/spec/simd.wast (hunk 1 above)
Comment (tlively) [reply test/spec/simd.wast:1054]: I think we wrote this one by hand as we were implementing SIMD interpretation. That was probably before there were any upstream spec tests.

**Comment by tlively** [comment]
### Merge activity

* **Apr 24, 2:06 PM EDT**: @tlively started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6529).

**Pull request merged**